File: backend/src/models/payment.mjs
import mongoose from "mongoose";

const paymentSlipSchema = new mongoose.Schema({
    studentId:{type:mongoose.Schema.Types.ObjectId,ref:'User',required:true},
    classId:{type:mongoose.Schema.Types.ObjectId, ref:'Class',required:true},
    amount:{type:Number},
    month:{type:String},
    slipFile:{type:String,required:true},
    status: {
    type: String,
    enum: ['pending', 'approved', 'rejected'],
    default: 'pending'
  },
    teacherId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' }, // who approved/rejected
  teacherComment: { type: String },
  submittedAt: { type: Date, default: Date.now },
  reviewedAt: { type: Date },
},{

 timestamps: true
  
});

export default mongoose.model('PaymentSlip', paymentSlipSchema);

File: frontend/lib/models/payment.dart
class PaymentSlip {
  final String? id;
  final String studentId;
  final String classId;
  final double amount;
  final String month;
  final String slipFile;

  PaymentSlip({
    this.id,
    required this.studentId,
    required this.classId,
    required this.amount,
    required this.month,
    required this.slipFile,
  });

  Map<String, dynamic> toJson() {
    return {
      'studentId': studentId,
      'classId': classId,
      'amount': amount,
      'month': month,
      'slipFile': slipFile,
    };
  }

  factory PaymentSlip.fromJson(Map<String, dynamic> json) {
    return PaymentSlip(
      id: json['_id'] as String? ?? json['id'] as String?,
      studentId:
          json['studentId'] as String? ??
          (json['student'] is Map ? json['student']['_id'] as String? : null) ??
          '',
      classId:
          json['classId'] as String? ??
          (json['class'] is Map ? json['class']['_id'] as String? : null) ??
          '',
      amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
      month: json['month'] as String? ?? '',
      slipFile: json['slipFile'] as String? ?? '',
    );
  }
}
